pkg/operator/status.go

```diff
@@ -23,6 +23,15 @@ const (
 	ReasonCheckAutoscaler   = "UnableToCheckAutoscalers"
 )
 
+// StatusReporterInterface is used to enable unit testing and does not
+// implement all methods of StatusReporter.
+type StatusReporterInterface interface {
```

Comment:
Can you make the interface private? It's fine to use it in unit tests. Though, nothing to be consumed outside of `pkg/operator` package.